jax/config.py

```diff
@@ -170,3 +170,9 @@ def __getattr__(self, name):
     int_env('JAX_TRACER_ERROR_NUM_TRACEBACK_FRAMES', 5),
     help='Set the number of stack frames in JAX tracer error messages.'
 )
+
+flags.DEFINE_bool(
+    'jax_check_tracer_leaks',
+    bool_env('JAX_CHECK_TRACER_LEAKS', False),
+    help='Turn on checking for leaked tracers as soon as a trace completes.'
```

Comment:
If this flag is also going to disable caching/shouldn't be left on outside actual debugging, it should probably say so here.

Reply:
Gooooood idea!